src/Dom/Document.php

```diff
@@ -258,6 +278,22 @@ public static function from_node( DOMNode &$node ) {
 		return $dom;
 	}
 
+	/**
+	 * Reset the internal optimizations of the Document object.
+	 *
+	 * This might be needed if you are doing an operation that causes the cached
+	 * nodes and XPath objects to point to the wrong document.
+	 *
+	 * @return self Reset version of the Document object.
+	 */
+	public function reset() {
```

Comment:
Does this need to be public?

Reply:
No, it's actually better to default to `private` for now, and only make it public if we actually now it is needed.

Reply:
Resolved via https://github.com/ampproject/amp-wp/pull/3758/commits/e006454b725fd4c1a0589af73211c0a82c5a97ed